src/client/pmix_client_get.c

```diff
@@ -615,7 +615,18 @@ static void _getnb_cbfunc(struct pmix_peer_t *pr,
                                 "pmix: get_nb searching for key %s for proc %s",
                                 cb->key, PMIX_NAME_PRINT(&proc));
             PMIX_GDS_FETCH_KV(rc, pmix_globals.mypeer, cb);
-            if (PMIX_SUCCESS == rc) {
+            if (PMIX_SUCCESS != rc) {
+               /* if we are both using the "hash" component, then the server's peer
+                * will simply be pointing at the same hash tables as my peer - no
+                * no point in checking there again */
+                if (0 != strcmp(pmix_globals.mypeer->nptr->compat.gds->name, pmix_client_globals.myserver->nptr->compat.gds->name)) {
+                    pmix_output_verbose(2, pmix_client_globals.get_output,
+                                        "pmix: get_nb searching for key %s for proc %s, - %s",
+                                        cb->key, PMIX_NAME_PRINT(&proc), pmix_client_globals.myserver->nptr->compat.gds->name);
+                    PMIX_GDS_FETCH_KV(rc, pmix_client_globals.myserver, cb);
+                }
+            }
+            else {
```

Comment:
errrr...I don't think that is correct. The problem now is that if the second fetch succeeds, you won't extract the value from the `cb` struct so it can be passed down. I believe you could make this work if you replace this "else" with `if (PMIX_SUCCESS == rc)`, though you might need to check it to be sure.

Reply:
Yeah, looks good - thx!